src/include/common/dedicated_thread_owner.h

```diff
@@ -16,35 +18,65 @@ namespace terrier {
  */
 class DedicatedThreadOwner {
  public:
+  virtual ~DedicatedThreadOwner() = default;
```

Comment:
I suspect we will need to write a lot more documentation detailing the sequence of how things work here. This module will probably need a design doc.